src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,46 +1,64 @@
+
 public class SalaryInfo {
-    /**
-     * <p>Реализуйте метод getSalaryInfo(String[] names, String[] data,
-     * String dateFrom, String dateTo)
-     * вычисляющий зарплату сотрудников. На вход методу подаётся 2 массива и 2 даты,
-     * определяющие период за который надо вычислить зарплату, первый массив содержит имена
-     * сотрудников организации, второй массив информацию о рабочих часах и ставке. Формат данных
-     * второго массива следующий: дата, имя сотрудника, количество отработанных часов,
-     * ставка за 1 час. Метод должен вернуть отчёт за период, который передали в метод
-     * (обе даты включительно) составленный по следующей форме: Отчёт за период
-     * #дата_1# - #дата_2# Имя сотрудника - сумма заработанных средств за этот период
-     * Создать класс-ошибку IllegalDateParametersException и сделать так, чтобы
-     * метод getSalaryInfo выбрасывал IllegalDateParametersException,
-     * если dateFrom > dateTo, с сообщнием "Wrong parameters"</p>
-     *
-     * <p>Пример ввода: date from = 01.04.2019 date to = 30.04.2019</p>
-     *
-     * <p>names:
-     * Сергей
-     * Андрей
-     * София</p>
-     *
-     * <p>data:
-     * 26.04.2019 Сергей 60 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50
-     * 26.04.2019 Андрей 3 200
-     * 26.04.2019 Сергей 7 100
-     * 26.04.2019 София 9 100
-     * 26.04.2019 Сергей 11 50</p>
-     *
-     * <p>Пример вывода:
-     * Отчёт за период 01.04.2019  - 30.04.2019
-     * Сергей - 1550
-     * Андрей - 600
-     * София - 900</p>
-     */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+    private static final DateTimeFormatter DATE_FORMATTER
+            = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data, String startDate,
+                                String endDate)
+            throws IllegalDateParametersException {
+        LocalDate start = LocalDate.parse(startDate, DATE_FORMATTER);
+        LocalDate end = LocalDate.parse(endDate, DATE_FORMATTER);
+
+        if (start.isAfter(end)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        StringBuilder report = new StringBuilder();
+        report.append("Отчёт за период ")
+                .append(startDate)
+                .append(" - ")
+                .append(endDate)
+                .append("\n");
+
+        int[] totalSalaries = new int[names.length];
+
+        for (String entry : data) {
+            String[] entryParts = entry.split(" ");
+            LocalDate entryDate = LocalDate.parse(entryParts[0], DATE_FORMATTER);
+
+            if (!entryDate.isBefore(start) && !entryDate.isAfter(end)) {
+                String employeeName = entryParts[1];
+                int hoursWorked = Integer.parseInt(entryParts[2]);
```

Comment:
Parsing integers from strings can throw a `NumberFormatException` if the string is not a valid integer. Consider adding error handling to manage such cases gracefully.